doc/jvm/xgboost4j_spark_tutorial.rst

```diff
@@ -153,6 +153,41 @@ Now, we have a DataFrame containing only two columns, "features" which contains
 "sepal length", "sepal width", "petal length" and "petal width" and "classIndex" which has Double-typed
 labels. A DataFrame like this (containing vector-represented features and numeric labels) can be fed to XGBoost4J-Spark's training engine directly.
 
+Dealing with missing values
+~~~~~~~~~~~~~~~~~~~~~~
+
+In the case that a feature column contains missing values for any reason (could be related to business logic / wrong data ingestion process / etc.), the user should decide on a strategy of how to handle it. 
+The choice of approach depends on the value representing 'missing' which fall into four different categories:
+1. 0.
+2. NaN.
+3. Null.
+4. any other value which is not mentioned in (1) / (2) / (3).
+
+We introduce the following approaches dealing with missing value and their fitting scenarios:
+
+1. Skip VectorAssembler (using setHandleInvalid = "skip") directly. Used in (2), (3).
+2. Keep it (using setHandleInvalid = "keep"), and set the "missing" parameter in XGBClassifier/XGBRegressor as the value representing missing. Used in (2) and (4).
+3. Keep it (using setHandleInvalid = "keep") and transform to other irregular values. Used in (3).
+4. Transform it to other values. Used in (1).
```

Comment:
"due to the fact that Spark's VectorAssembler transformer only accepts 0 as the missing values. This issue creates a problem when the user has 0 as meaningful value, plus there are enough 0's to use SparseVector.	

 In case the dataset is represented by a DenseVector, the 0 is kept."

I mean this paragraph from your original commits is informative, you can keep it

Reply:
ah I see. I will add it as well then.

Reply:
I changed accordingly based on your current structure

Reply:
OK, thanks for the changes from your side as well. When expected to see it in tutorial, @CodingCat ?
Thank you

Reply:
The tutorial will be available right after this PR is merged.

Reply:
Thank u @hcho3 . I saw failures on Travis CI - should I ignore it for now @CodingCat ?

Reply:
@Daniel8hen The failed test is most likely due to flaky internet connection. I re-started the test.

Reply:
Thx @hcho3 . So merge will be automatically once all tests and builds are done?

Reply:
Yes, I'll merge it. @CodingCat Are we good with what we have here now?

Reply:
yes, thanks